linux-arm64 build error

I just build marksman with README, and I get this error message ... 

system : Ubuntu 22.04 arm64 [chroot in an Android machine]
> /home/zoran/.dotnet/sdk/6.0.406/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(90,5): error NETSDK1083: The specified RuntimeIdentifier 'linux-' is not recognized. [/tmp/marksman/Marksman/Marksman.fsproj]

`uname -p` returns `unknown` on both x86_64 and aarch64 linux on my side (`-p` is marked as non-portable)

`uname -m` however properly returns `x86_64` and `aarch64` on both respective systems.

I tried fixing the Makefile but `make publish` gave me a .net error I'm not familiar with:

```
dotnet publish -c Release -r linux-arm64 --self-contained true \
        -p:PublishSingleFile=true \
        -p:PublishTrimmed=true \
        -p:DebugType=embedded \
        -p:EnableCompressionInSingleFile=true \
        Marksman/Marksman.fsproj
MSBuild version 17.3.2+561848881 for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.
  670524e-dirty
/nix/store/ahyyjjmnzsdmr7q55xc7hsp23jpg6yh9-dotnet-sdk-6.0.405/sdk/6.0.405/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.RuntimeIdentifierInference.targets(157,5): error NETSDK1067: Self-contained applications are required to use the application host. Either set SelfContained to false or set UseAppHost to true. [/home/mrene/marksman/Marksman/Marksman.fsproj]
make: *** [Makefile:72: publish] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -20,20 +20,37 @@ ifeq ($(OS),Windows_NT)
 	endif
 else
 	UNAME_S := $(shell uname -s)
+	TARGET := $(shell uname -m)
 	ifeq ($(UNAME_S),Linux)
 		OS_ID := linux
 	endif
+
 	ifeq ($(UNAME_S),Darwin)
 		OS_ID := osx
+		ifeq ($(TARGET),arm64)
+			# Non-versioned OSX targets do not support arm64
+			OS_ID := "osx.11.0"
+			ARCH_ID := arm64
+		endif
 	endif
-	UNAME_P := $(shell uname -p)
-	ifeq ($(UNAME_P),x86_64)
+
+	ifeq ($(TARGET),x86_64)
 		ARCH_ID := x64
 	endif
-	ifneq ($(filter %86,$(UNAME_P)),)
-		ARCH_ID := x64
+
+	ifeq ($(TARGET),x86)
+		ARCH_ID := x86
 	endif
-	ifneq ($(filter arm%,$(UNAME_P)),)
+
+	ifeq ($(TARGET),arm)
+		ARCH_ID := arm
+	endif
+	
+	ifeq ($(TARGET),arm64)
+		ARCH_ID := arm64
+	endif
+
+	ifeq ($(TARGET),aarch64)
 		ARCH_ID := arm64
 	endif
 endif
@@ -71,6 +88,7 @@ publish:
 		-p:PublishTrimmed=true \
 		-p:DebugType=embedded \
 		-p:EnableCompressionInSingleFile=true \
+		-p:UseAppHost=true \
 		Marksman/Marksman.fsproj
 		
 .PHONY: publishTo
@@ -81,6 +99,7 @@ publishTo:
 		-p:DebugType=embedded \
 		-p:EnableCompressionInSingleFile=true \
 		-p:VersionString=$(VERSIONSTRING) \
+		-p:UseAppHost=true \
 		Marksman/Marksman.fsproj \
 		-o $(DEST)
 		

